src/server.rs

```diff
@@ -250,13 +250,22 @@ async fn run_server(
         secp256k1::PublicKey::from_secret_key(&Secp256k1::new(), &core_secret_key);
     let host_id = hex::encode(core_public_key.serialize());
     tokio::spawn(print_host_token(jwt_secret, host_id.clone()));
+    let uri_parsed = url::Url::parse(&mq_uri)?;
+    let mq: Arc<dyn MQ> = if uri_parsed.scheme().starts_with("redis") {
```

Comment:
while this is totally fine, just curious: why are we switching from single ownership to cloned handler? Any particular reason?